twill-core/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -171,9 +168,27 @@ public String getKafkaZKConnect() {
   /**
    * Returns the minimum heap ratio ({@link Configs.Keys#HEAP_RESERVED_MIN_RATIO}) based on the given configuration.
    */
-  private double getMinHeapRatio(Map<String, String> config) {
-    return config.containsKey(Configs.Keys.HEAP_RESERVED_MIN_RATIO) ?
-      Double.parseDouble(config.get(Configs.Keys.HEAP_RESERVED_MIN_RATIO)) :
-      Configs.Defaults.HEAP_RESERVED_MIN_RATIO;
+  private double getMinHeapRatio(@Nullable Map<String, String> config, double defaultValue) {
+    if (config == null) {
```

Comment:
I think this would be easier to read, and it would give better errors, as follows:
```
if (config == null || !config.containsKey(...)) {
  return defaultValue;
} 
try {
  double ratio = Double.parseDouble(config.get(...));
} catch (NumberFormatException) {
  // either warn or throw a more informative exn (including runnable name and the bad value)
}
if (ratio <= 0d) {
  // either warn or throw a more informative exn (including runnable name and the bad value)
}
return ratio;
```